src/commands/generate/models.ts

```diff
@@ -140,95 +145,103 @@ export default class Models extends Command {
-      fileGenerator = new PythonFileGenerator();
-      break;
-    case Languages.rust:
-      fileGenerator = new RustFileGenerator();
-      break;
-    case Languages.csharp:
-      if (namespace === undefined) {
-        throw new Error('In order to generate models to C#, we need to know which namespace they are under. Add `--namespace=NAMESPACE` to set the desired namespace.');
-      }
-
-      fileGenerator = new CSharpFileGenerator({
-        presets: csharpAutoImplement ? [
-          {
-            preset: CSHARP_DEFAULT_PRESET,
+      case Languages.typescript:
+        if (tsIncludeComments) { presets.push(TS_DESCRIPTION_PRESET); }
+        if (tsExampleInstance) {
+          presets.push({
+            preset: TS_COMMON_PRESET,
             options: {
-              autoImplementedProperties: true
+              example : true
             }
-          }
-        ] : [],
-        collectionType: csharpArrayType as 'Array' | 'List'
-      });
-      
-      fileOptions = {
-        namespace
-      };
-      break;
-    case Languages.golang:
-      if (packageName === undefined) {
-        throw new Error('In order to generate models to Go, we need to know which package they are under. Add `--packageName=PACKAGENAME` to set the desired package name.');
-      }
-      fileGenerator = new GoFileGenerator();
-      fileOptions = {
-        packageName
-      };
-      break;
-    case Languages.java:
-      if (packageName === undefined) {
-        throw new Error('In order to generate models to Java, we need to know which package they are under. Add `--packageName=PACKAGENAME` to set the desired package name.');
-      }
-      fileGenerator = new JavaFileGenerator();
-      fileOptions = {
-        packageName
-      };
-      break;
-    case Languages.javascript:
-      fileGenerator = new JavaScriptFileGenerator();
-      break;
-    case Languages.dart:
-      if (packageName === undefined) {
-        throw new Error('In order to generate models to Dart, we need to know which package they are under. Add `--packageName=PACKAGENAME` to set the desired package name.');
-      }
-      fileGenerator = new DartFileGenerator();
-      fileOptions = {
-        packageName
-      };
-      break;
-    case Languages.kotlin:
-      if (packageName === undefined) {
-        throw new Error('In order to generate models to Kotlin, we need to know which package they are under. Add `--packageName=PACKAGENAME` to set the desired package name.');
-      }
-      fileGenerator = new KotlinFileGenerator();
-      fileOptions = {
-        packageName
-      };
-      break;
-    default:
-      throw new Error(`Could not determine generator for language ${language}, are you using one of the following values ${possibleLanguageValues}?`);
+          })
+        }
+        if (tsJsonBinPack) {
+          presets.push({
+            preset: TS_COMMON_PRESET,
+            options: {
+              marshalling: true
+            }
+          },
```

Comment:
Do you see that the same preset are added twice now, just with slightly different options? 🙂 

This should not be the case, it can only be added once, and then the options should change slightly based on the parameters 🙂

Reply:
It is not getting added twice even if there are both flags in the command of `tsExampleInstance` and `tsJsonBinPack`. I think Modelina has the support for this where the same preset is added with multiple values of options then it gets combined to give the output. Just the order of the functions generated is getting changed somewhat rest is the same for me.

Reply:
Another approach would be to create an object variable 
```javascript
const options={
    example : tsExampleInstace ,
    marshalling : tsMarshalling
}

presets.push({
    preset: TS_COMMON_PRESET,
    options:options
})
```
and then use this while pushing the presets which can also resolve #460

Reply:
Ahh you right, it's allowed as long as the options don't overlap or default behavior adds some stuff. Because otherwise it will be added twice 😄